src/linux/ldcache.cpp

```diff
@@ -146,33 +162,49 @@ Try<vector<Entry>> parse(const string& path)
 
   const char* data = buffer->data();
 
-  // Grab a pointer to the old format header (for verification of
-  // HEADER_MAGIC_OLD later on). Then jump forward to the location of
-  // the new format header (it is the only format we support).
-  HeaderOld* headerOld = (HeaderOld*)data;
-  data += sizeof(HeaderOld);
-  if (data >= buffer->data() + buffer->size()) {
+  HeaderNew* headerNew = (HeaderNew*)data;
+  if (data + sizeof(HeaderNew) >= buffer->data() + buffer->size()) {
     return Error("Invalid format");
   }
 
-  data += headerOld->libraryCount * sizeof(EntryOld);
-  if (data >= buffer->data() + buffer->size()) {
-    return Error("Invalid format");
-  }
+  if (strncmp(headerNew->magic,
+        HEADER_MAGIC_NEW,
+        sizeof(HEADER_MAGIC_NEW) - 1) != 0) {
```

Comment:
Not yours, the indent here seems not correct, it should be:

```
  if (strncmp(headerNew->magic,
      HEADER_MAGIC_NEW,
      sizeof(HEADER_MAGIC_NEW) - 1) != 0) {
```

See [this code](https://github.com/apache/mesos/blob/1.11.0/src/slave/containerizer/mesos/containerizer.cpp#L599:L601) as an example. Ditto for the other `strncmp` below.